todo_interface.rb

```diff
@@ -0,0 +1,165 @@
+require_relative 'todo_item_class.rb'
+require_relative 'todo_list_class.rb'
+require 'colorize'
+require 'date'
+
+
+def input_date
+	while(true)
+		puts "Now enter the due date in the form \"dd-mm-yyyy\": "
+		date_input = gets.chomp
+		d, m, y = date_input.split('-')
+		if Date.valid_date? y.to_i, m.to_i, d.to_i
+			begin
+			 	 return Date.parse(date_input)
+			 rescue Exception => e
+			 end
+		end
+		puts "Oops! Invalid date -- try again!".red
+	end
+end
+
+def input_priority
+	while(true)
+		puts "Now enter priority. Remember: 1 = high, 2 = medium, 3 = low"
+		priority = gets.chomp
+		if (1..3).include? priority.to_i
+			return priority.to_i
+		end
+		puts "Oops! Invalid priority -- try again!".red
+	end
+end
+
+
+def initialize_list
+	puts "Welcome to your To-Do List! Let's get started!"
+
+	#initialize list
+	puts "To-Do List name: "
+	list_name = gets.chomp
+	list = ToDoList.new(list_name)
+	puts "Created To-Do List with name #{list.name}".blue
+	return list
+end
+
+def print_list(list)
+	sort_type = ""
+	completed = ""
+	while(true)
+		puts "Would you like to sort your list by priority or due date? [p/d]"
+		sort_type = gets.chomp
+		if sort_type != 'p' && sort_type != 'd'
+			puts "Please enter 'p' for priority or 'd' for due date".red
+			next
+		end
+		puts "Would you like the list to include items that are completed? [y/n]"
+		completed = gets.chomp
+		if completed != 'y' && completed != 'n'
+			puts "Please enter 'y' to include completed items, or 'n' to not include them".red
+			next
+		end
+		completed = completed == 'y' ? true : false
+		break
+	end
+	puts "\nTODO LIST: #{list.name}".blue
```

Comment:
I think lines 64-77 would make a great `print` method on your list, and would make this `print_list` method a bit more concise.